### ISSUE TITLE
Mistake in "web3-eth-subscribe.html#subscribe-newblockheaders" Example

### ISSUE THREAD

**Issue by Remi-Burgel** [ethereum/web3.js#1367]
Hi,
I guess I could modify this myself but I'm a newbe on github. On [this](https://web3js.readthedocs.io/en/1.0/web3-eth-subscribe.html#subscribe-newblockheaders) page, there is a mistake in the example.

The example say : 
```
var subscription = web3.eth.subscribe('newBlockHeaders', function(error, result){
    if (!error)
        console.log(error);
})
.on("data", function(blockHeader){
});
```

It means that if you have NO error your print the error. It's silly but weird to let it there.

**Comment by BransonGitomeh** [comment]
my issue with the example is that you have two events, the call back and the on data event

correct is

```javascript
web3.eth
      .subscribe("newBlockHeaders")
      .on("data", async transaction => {
        console.log({ transaction });
      })
      .on("error", function(err) {
        console.log({ err });
      });
```

**Comment by frozeman** [comment]
Thanks, these are all options shown, you can use either the `error` event OR the callback.

**Comment by frozeman** [comment]
fixed in #1876
